### ISSUE TITLE
Nesting of translations breaks Testing

### ISSUE THREAD

**Issue by abhilashlr** [ember-intl/ember-intl#1035]
Since [nesting (similar to this)](https://www.i18next.com/translation-function/nesting) is not supported by ember-intl yet, the alternative is to define a translation json as:
```
{
  "reuseText": "{count, plural, =1 {Name} other {Names}}",
  "secondKey: "{reuseText} has Hello world"
}
```
and in the template we define the usage as `{{t "secondKey" reuseText=(t "reuseText" count=2)}}`. This isn't going to throw an error. Instead, the `reuseText` key is translated to proper text and applied as a param to the `secondKey`.

But the same template when running in its test case throws an error similar to [this](https://github.com/ember-intl/ember-intl/issues/616).

After debugging, the test cases scenario, I figured out that the `stringify` method makes an additional escape character for `double quotes: "`. And the same value is now applied to the first translation as a value to `reuseText`. But since the intl.message.parser cannot translate a key that contains a `\`, the error of #616 comes up - but only in test cases. The reason being, ember-intl's test-support helper overrides the [missing-message translation util](https://github.com/ember-intl/ember-intl/blob/master/addon-test-support/setup-intl.js#L26).

1. One possible solution is to fix this to replace even the escaped double quotes with a simple doubt quotes in [here](https://github.com/ember-intl/ember-intl/blob/master/addon-test-support/-private/serialize-translation.js#L23).

2. Or, if nesting translations is itself a part of the ember-intl addon, the solution gets simpler where we could define the translation something like the following similar to i18next:
```
{
  "reuseText": "{count, plural, =1 {Name} other {Names}}",
  "secondKey: "$t(reuseText) has Hello world"
}
```

3. Or, is it a mistake to do a translation and then apply that string value to another key?

4. The other possible solution is to rewrite all the strings which are similar to `reuseText` as it's value manually in all the places which are again a tedious manual process like the following:
```
{
  "reuseText": "{count, plural, =1 {Name} other {Names}}",
  "secondKey: "{count, plural, =1 {Name} other {Names}} has Hello world"
}
```
5. Or, is there another totally different solution to this?

- [x] I am on the [latest](https://github.com/ember-intl/ember-intl/releases/latest) ember-intl version
- [x] I have searched the [issues](https://github.com/ember-intl/ember-intl/issues) of this repo and believe that this is not a duplicate

**Comment by abhilashlr** [comment]
@jasonmit The PR #1036 addresses fixing this issue by adopting solution 1.

**Comment by jasonmit** [comment]
> and in the template we define the usage as {{t "secondKey" reuseText=(t "reuseText" count=2)}}

Seems like the best approach for now.

>Or, if nesting translations is itself a part of the ember-intl addon, the solution gets simpler where we could define the translation something like the following similar to i18next:

I would be willing to adopt this strategy.  The build tooling can combine the translations together.

**Comment by jasonmit** [comment]
@buschtoens I know you did a lot of work on the testing story, what are your thoughts?

I feel we should fix the bug even if we add first class support for that reflecting behavior.

**Comment by abhilashlr** [comment]
> I feel we should fix the bug even if we add first class support for that reflecting behavior.

Not sure if #1036 is the right way to fix it, what are your thoughts @jasonmit @buschtoens?

**Comment by jasonmit** [comment]
@abhilashlr I don't believe so as it's going to interfere with translations with quotes.

**Comment by abhilashlr** [comment]
> @abhilashlr I don't believe so as it's going to interfere with translations with quotes.

@jasonmit agree to your point, but this would only replace the `\"` in the translation key, something defined like:
```
"translation\"s_key": "Something"
```
where the output would be something like `...translation)_key...`

**Comment by jasonmit** [comment]
Okay, within the key isn't so much of a concern then.

> EDIT: I just realised that the PR replaced \" with a ) instead, moved it to replace with "

Yeah, that part confused me as well :)

I still want to defer this to @buschtoens since I believe he wrote most of the test support logic.  It's definitely a bug none the less.  If he doesn't have the cycles, I can take it over!

**Comment by abhilashlr** [comment]
> Yeah, that part confused me as well :)

Sorry about that! 🤦‍♂ 

> I still want to defer this to @buschtoens since I believe he wrote most of the test support logic. It's definitely a bug none the less. If he doesn't have the cycles, I can take it over!

Sure, thanks for that @jasonmit

**Comment by buschtoens** [comment]
I'll take a look!

**Comment by abhilashlr** [comment]
> I'll take a look!

@buschtoens did you get a chance to look at this?

**Comment by jasonmit** [comment]
Will release a 4.x patch soon.  Closing, thanks!

**Comment by jasonmit** [comment]
Released as v4.3.2

Thanks!
